airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -61,17 +63,25 @@ def discover(self, logger: AirbyteLogger, config_container: ConfigContainer) ->
         return AirbyteCatalog(streams=client.get_streams())
 
     def read(
-        self, logger: AirbyteLogger, config_container: ConfigContainer, catalog_path, state=None
+            self, logger: AirbyteLogger, config_container: ConfigContainer, catalog_path, state_path: str = None
     ) -> Generator[AirbyteMessage, None, None]:
         client = self._client(config_container)
 
+        if state_path:
+            logger.info("Starting sync with provided state file")
+            state_obj = json.loads(open(state_path, "r").read())
+        else:
+            logger.info("No state provided, starting fresh sync")
+            state_obj = {}
+
+        state = defaultdict(dict, state_obj)
         catalog = ConfiguredAirbyteCatalog.parse_obj(self.read_config(catalog_path))
 
         logger.info("Starting syncing mailchimp")
         for configured_stream in catalog.streams:
             stream = configured_stream.stream
-            for record in self._read_record(client=client, stream=stream.name):
-                yield AirbyteMessage(type=Type.RECORD, record=record)
+            for record in self._read_record(client=client, stream=stream.name, state=state):
+                yield record
```

Comment:
you're changing the type you're yielding, is that expected?

Reply:
yes, because the endpoints are now yielding either a state or a record message